geode-core/src/main/java/org/apache/geode/internal/process/FileProcessController.java

```diff
@@ -117,13 +117,21 @@ private String status(final File workingDir, final String statusRequestFileName,
     File statusFile = new File(workingDir, statusFileName);
     AtomicReference<String> statusRef = new AtomicReference<>();
 
-    ControlRequestHandler statusHandler = () -> {
-      // read the statusFile
-      StringBuilder lines = new StringBuilder();
-      try (BufferedReader reader = new BufferedReader(new FileReader(statusFile))) {
-        reader.lines().forEach(lines::append);
-      } finally {
-        statusRef.set(lines.toString());
+    ControlRequestHandler statusHandler = new ControlRequestHandler() {
+      @Override
+      public void handleRequest() throws IOException {
+        handleRequest(null);
+      }
+
+      @Override
+      public void handleRequest(FileReader fileReader) throws IOException {
+        // read the statusFile
+        StringBuilder lines = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(fileReader)) {
```

Comment:
According BufferReader(FileReader) implementation; it will thrown NPE when null arg is passed.

Reply:
Yes, that isn't good. I think I've found a better way to fix this bug without doing this.